hal/src/i2c.rs

```diff
@@ -0,0 +1,412 @@
+//! # Inter-Integrated Circuit (I2C) bus
+
+use core::convert::TryFrom;
+
+use crate::{
+    embedded_hal::blocking::i2c::{Read, Write, WriteRead},
+    pac::{rcc::ccipr::I2C3SEL_A, I2C1, I2C2, I2C3, RCC},
+    rcc::{sysclk_hz, pclk1_hz},
+};
+
+use embedded_time::{fixed_point::FixedPoint, rate::*};
+
+/// I2C error
+#[derive(Debug)]
```

Comment:
```suggestion
#[cfg_attr(feature = "defmt", derive(defmt::Format))]
```

To maintain optional compatibility with `defmt::unwrap!`